### ISSUE TITLE
Nodemon can't shutdown express 4 app when using required version of nodemon using gulp

### ISSUE THREAD

**Issue by Artsu** [remy/nodemon#350]
I'm trying to use Nodemon from a gulp task. First I tried to use gulp-nodemon but ran into <code>Error: listen EADDRINUSE</code> error. After a while I decided to try pure nodemon after which I noticed the same problem seems to exist there too.

My server.js is basically (cleaned up to minimal version):

``` sh
var app = express();

var server = app.listen(config.port, function () {
  console.log('Express server listening on port %d in %s mode', config.port, app.get('env'));
});
```

And on gulpfile I have (cleaned up to minimal version):

``` sh
gulp.task('run-backend', function () {
  nodemon({script : 'server.js', cwd: __dirname});
});
```

When I run <code>nodemon server.js</code> everything works just fine. However when I do <code>gulp serve</code> it starts the server and nodemon watch but every time nodemon tries to restart the server the following error is thrown as the server never gets shutdown:

``` sh
events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: listen EADDRINUSE
    at errnoException (net.js:901:11)
    at Server._listen2 (net.js:1039:14)
    at listen (net.js:1061:10)
    at Server.listen (net.js:1127:5)
    at Function.app.listen (*****/node_modules/express/lib/application.js:532:24)
    at Object.<anonymous> (*****/server.js:39:18)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
});
```

I'm using <code>express 4.2.0</code> and <code>nodemon 1.2.0</code>

Is required version supported anymore? I suppose I could just run gulp and nodemon on separate consoles but it would be nice to do everything with just one command.

**Comment by KarandikarMihir** [comment]
How weird is that! I just replaced `__dirname` with `process.cwd()` in my express app and everything started to work. I'm not sure what is causing this issue or how `process.cwd()` solves it.

**Comment by remy** [comment]
I believe this is also fixed with nodemon@1.12.7 and the recent `fork` support. Happy to reopen (or create a new issue if not).
